src/build-data/oids.txt

```diff
@@ -25,6 +25,20 @@
 1.3.6.1.4.1.2.267.10.3.3 = Kyber-768-90s-r3
 1.3.6.1.4.1.2.267.10.4.4 = Kyber-1024-90s-r3
 
+# Dilithium OIDs are currently in Botan's private arc
+1.3.6.1.4.1.25258.1.9.1 = Dilithium-r3/4x4-randomized
+1.3.6.1.4.1.25258.1.9.2 = Dilithium-r3/4x4-deterministic
+1.3.6.1.4.1.25258.1.9.3 = Dilithium-r3/6x5-randomized
+1.3.6.1.4.1.25258.1.9.4 = Dilithium-r3/6x5-deterministic
+1.3.6.1.4.1.25258.1.9.5 = Dilithium-r3/8x7-randomized
+1.3.6.1.4.1.25258.1.9.6 = Dilithium-r3/8x7-deterministic
+1.3.6.1.4.1.25258.1.10.1 = Dilithium-AES-r3/4x4-randomized
+1.3.6.1.4.1.25258.1.10.2 = Dilithium-AES-r3/4x4-deterministic
+1.3.6.1.4.1.25258.1.10.3 = Dilithium-AES-r3/6x5-randomized
+1.3.6.1.4.1.25258.1.10.4 = Dilithium-AES-r3/6x5-deterministic
+1.3.6.1.4.1.25258.1.10.5 = Dilithium-AES-r3/8x7-randomized
+1.3.6.1.4.1.25258.1.10.6 = Dilithium-AES-r3/8x7-deterministic
```

Comment:
OK sorry to bring this up now at kind of the last minute, but:

Is it really sensible to make a distinction between the randomized and deterministic keys, at the OID/key format level?

AIUI, the keys would have the exact same encoding, and additionally signatures generated by either deterministic or randomized are indistinguishable to the verifier. Also I notice draft-uni-qsckeys-dilithium makes no such distinction between the keys.

I feel like we could significantly simplify the code by removing the "flavor" mechanism entirely except specifically in the context of generating the signatures. There, the behavior could be chosen by the parameters provided to `PK_Signer` (which are currently ignored/unused). Eg 

```
PK_Signer sign_with_randomness(*dilithium_key, "Randomized");
PK_Signer sign_without_randomness(*dilithium_key, "Deterministic");
PK_Signer sign_with_some_default(*dilithium_key); // don't care (probably use deterministic by default?)
```

Reply:
Sounds reasonable to me. I don't expect this to be a complicated change, even that late in the implementation.

Reply:
Done in https://github.com/randombit/botan/pull/2973/commits/f61ee4c3bb04784970cc3d83cce8d0da33a18b5d.

We should definitely remember to squash those commits before merging...